test/server/http/admin_test.cc

```diff
@@ -293,5 +294,121 @@ TEST(PrometheusStatsFormatter, FormattedTags) {
   EXPECT_EQ(expected, actual);
 }
 
+TEST(PrometheusStatsFormatter, MetricNameCollison) {
+
+  //Create two counters and two gauges with each pair having the same name, 
+  //but having different tag names and values.  
+  //`statsAsPrometheus()` should return two implying it found two unique stat names
+
+  Stats::HeapRawStatDataAllocator alloc;
+  std::list<Stats::CounterSharedPtr> counters;
+  std::list<Stats::GaugeSharedPtr> gauges;
+
+ {
+  std::string name = "cluster.test_cluster_1.upstream_cx_total";
+  std::vector<Stats::Tag> cluster_tags;
+  Stats::Tag tag = {"a.tag-name", "a.tag-value"};
+  cluster_tags.push_back(tag);
+  Stats::RawStatData* data = alloc.alloc(name);
+  Stats::CounterSharedPtr c(
+      new Stats::CounterImpl(*data, alloc, std::string(name), std::move(cluster_tags)));
+  counters.push_back(c);
+  }
+
+  {
+  std::string name = "cluster.test_cluster_1.upstream_cx_total";
+  std::vector<Stats::Tag> cluster_tags;
+  Stats::Tag tag = {"another_tag_name", "another_tag-value"};
+  cluster_tags.push_back(tag);
+  Stats::RawStatData* data = alloc.alloc(name);
+  Stats::CounterSharedPtr c(
+      new Stats::CounterImpl(*data, alloc, std::string(name), std::move(cluster_tags)));
+  counters.push_back(c);
+  }
+
+  {
+  std::vector<Stats::Tag> cluster_tags;
+  std::string name = "cluster.test_cluster_2.upstream_cx_total";
+  Stats::Tag tag = {"another_tag_name_3", "another_tag_3-value"};
+  cluster_tags.push_back(tag);
+  Stats::RawStatData* data = alloc.alloc(name);
+  Stats::GaugeSharedPtr g(
+      new Stats::GaugeImpl(*data, alloc, std::string(name), std::move(cluster_tags)));
+  gauges.push_back(g);
+  }
+
+ {
+  std::vector<Stats::Tag> cluster_tags;
+  std::string name = "cluster.test_cluster_2.upstream_cx_total";
+  Stats::Tag tag = {"another_tag_name_4", "another_tag_4-value"};
+  cluster_tags.push_back(tag);
+  Stats::RawStatData* data = alloc.alloc(name);
+  Stats::GaugeSharedPtr g(
+      new Stats::GaugeImpl(*data, alloc, std::string(name), std::move(cluster_tags)));
+  gauges.push_back(g);
+  }
+
+  Buffer::OwnedImpl response;
+  EXPECT_EQ(2UL, PrometheusStatsFormatter::statsAsPrometheus(counters, gauges, response));
+}
+
+TEST(PrometheusStatsFormatter, UniqueMetricName) {
+
+  //Create two counters and two gauges with each pair having the different name, 
```

Comment:
nit - can we rephrase to: "Create two counters and two gauges, all with unique names. statsAsPrometheus() should return four implying it found four unique stat names."

Reply:
done thanks